sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulProcessorHandleImpl.scala

```diff
@@ -185,6 +202,16 @@ class StatefulProcessorHandleImpl(
     timerState.listTimers()
   }
 
+  /**
+   * Performs the user state cleanup based on assigned TTl values. Any state
+   * which is expired will be cleaned up from StateStore.
+   */
+  def doTtlCleanup(): Unit = {
+    ttlStates.forEach { s =>
```

Comment:
Just curious - this would be used to perform eviction for all types in the future - correct ?

Reply:
correct. All types should have a associated ttlState.